src/descendPeak.cpp

```diff
@@ -0,0 +1,143 @@
+// Pascal Maas - p.maas@lacdr.leidenuniv.nl
+
+#include <Rcpp.h>
+using namespace Rcpp;
+using namespace std;
+
+double weightedMean(NumericVector mzs, NumericVector ints) {
+    // Get the total intensity
+    double intensity = sum(ints);
+
+    // Calculate the weighted mass
+    int n = ints.length();
+    double total = 0;
+    for (int i = 0; i < n; i++) {
+        total += mzs[i] * ints[i] / intensity;
+    }
+    return total;
+}
+
+double descend(int centroid, NumericVector intensities,
+               NumericVector masses, double signalRatio) {
+
+    // Maximum deviation from the centroid index
+    int maxK = 30;
```

Comment:
Depending on the solution of the spectra an window of +/- 30 may be to large/small. I don't like to hardcode window sizes. I would suggest to make it an argument as well.

Reply:
Agreed, I will push an update that adds the `maxK` parameter to the function arguments.